File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    
    temp = []
    for i in s:
        if i == "(":
            temp.append(i)
        else:
            
            if len(temp) == 0 and i == ")":
                return False
            
            if len(temp) > 0 and temp[-1] == ")":
                return False
            
            if temp[-1] == "(":
                temp.pop()
                
            
    return len(temp) == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> stack = new Stack<>();
        
        for (char c: s.toCharArray()) {
            switch (c) {
                case '(':
                    stack.push(c);
                    break;
                case ')':
                    if (stack.isEmpty()) return false;
                    stack.pop();
                    break;
            }
        }
        
        return stack.isEmpty() ? true : false;
    }
}